framework/src/main/java/io/github/kgress/scaffold/webdriver/WebDriverManager.java

```diff
@@ -181,11 +181,20 @@ private void openUrl(WebDriverWrapper webDriver, String url) {
      */
     private MutableCapabilities getDesiredCapabilities() {
         MutableCapabilities browserOptions;
-
+        boolean headless = false;
+        if (runType == HEADLESS) {
+            headless = true;
+        }
         if (runType != UNIT) {
             switch (browserType) {
                 case Chrome:
-                    browserOptions = new ChromeOptions().setAcceptInsecureCerts(true);
+                    if (headless) {
+                        log.info("Setting up headless browser with maximized screen.");
+                        browserOptions = new ChromeOptions().setAcceptInsecureCerts(true)
+                                .addArguments("--headless").addArguments("--window-size=1440x5000");
```

Comment:
Instead of configuring a window size here, could we instead pass `.addArguments("--start-maximized"`? This will help in the event we're sending the testing to a vm that doesn't support the resolution of 1440x5000. Navigation to elements shouldn't be impacted by this change.

Reply:
Oh, better yet, it looks like ChromeOptions has a [`setHeadless()` function](https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/chrome/ChromeOptions.html#setHeadless-boolean-). It's unfortunate it doesn't have a maximize built into it, though.

Reply:
This also begs the question, should we go ahead and configure multiple browsers with Headless functionality? Do we care about having that expanded support?

I think including only Chrome support for now is fine. We can always come back and refactor later with additional browsers.

Reply:
This is updated to use setHeadless().   Getting a bunch of errors when I pull this into my current project though.   I think there are upstream changes you made thats causing things to fail.

Reply:
set-maximized didn't work on chrome mac, just fyi.